File: Find_and_Replace_Pattern.py

words = ["abc","deq","mee","aqq","dkd","ccc"]
pattern = "abb"

# words = ["a","b","c"]
# pattern = "a"


def nextGreaterElement(words,pattern):
    def get_pattern(w):
        res = []
        d = {}
        i=0
        for w1 in w:
            if w1 in d:
                res.append(d[w1])
            else:
                i+=1
                d[w1] = i
                res.append(i)

        return res

    p = get_pattern(pattern)

    out = []
    for w2 in words:
        if get_pattern(w2) == p:
            out.append(w2)

    return out


print(nextGreaterElement(words,pattern))

File: Find_and_Replace_Pattern/Find_and_Replace_Pattern.go
package main

import (
	"fmt"
)

func main() {
	words := []string{"abc", "deq", "mee", "aqq", "dkd", "ccc"}
	pattern := "abb"

	fmt.Println(findAndReplacePattern(words, pattern))
}

func findAndReplacePattern(words []string, pattern string) []string {

	getPattern("deq", pattern)
	return words
}

func getPattern(w string, pattern string) {
	m1, m2 := make(map[string]string), make(map[string]string)

	for i := 0; i < len(pattern); i++ {
		b1 := w[i]
		b2 := w[i]
		if _, ok := m1[string(b1)]; !ok {
			m1[string(b1)] = string(b2)
		}
		if _, ok := m2[string(b2)]; !ok {
			m2[string(b2)] = string(b1)
		}

	}

	fmt.Println(m1, m2)

}
